### PR TITLE
recent posts component

### LINKED ISSUE #3476: "Recent Posts" react component
See https://invis.io/NB7VM1GSG#/242718039_Dashboard_with_Discussions_card

In the interests of MVP, we can omit some fields/content that appear in the mock-up.

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #3476 

#### What's this PR do?

This improves the existing 'discussions' card to implement the mocks, by hitting the OD backend to fetch the frontpage and then rendering it.

#### How should this be manually tested?

Make sure you have all the stuff set up for OD and MM to talk to each other. You should also have `FEATURE_OPEN_DISCUSSIONS_POST_UI` set to `True`. Then you should see something like this on the dashboard:

obama

If you don't set the `POST_UI` feature flag or you don't set the redirect URL env var you shouldn't see the card. If you do, you should!

### CODE REVIEW

**Comment by pdpinch** [comment]
I added the Needs Review label. If that was premature, you can remove it.

File: static/js/lib/auth.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// @flow
+import "isomorphic-fetch"
+
+export const refreshAuthToken = () => {
```
Comment (noisecapella) [inline static/js/lib/auth.js]: Can you write tests for these two functions?

Comment (aliceriot) [reply static/js/lib/auth.js]: :woman_facepalming:

Comment (aliceriot) [reply static/js/lib/auth.js]: I even had that in my checklist when I was working and just totally forgot, haha

**Comment by aliceriot** [comment]
@noisecapella I added a few tests that should cover the functions in `lib/auth.js`

File: static/js/lib/auth.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+export const fetchJSONWithAuthToken = async (url: string) => {
+  let response = await fetch(url, { credentials: "include" })
+
+  if (response.status === 401) {
+    await refreshAuthToken()
+    response = await fetch(url, { credentials: "include" })
+  }
+
+  if (response.status < 200 || response.status > 300) {
```
Comment (noisecapella) [inline static/js/lib/auth.js]: Can you change the expression to be `>= 300` just to be clear that `300` is unexpected?

**Comment by aliceriot** [comment]
@noisecapella I fixed that check and updated the relevant test

**Pull request merged**